WordPress/Classes/Utility/PhotonImageURLHelper.m

```diff
@@ -2,12 +2,117 @@
 
 @implementation PhotonImageURLHelper
 
+static const NSUInteger DefaultPhotonImageQuality = 80;
+static const NSInteger MaxPhotonImageQuality = 100;
+static const NSInteger MinPhotonImageQuality = 1;
+
 + (NSURL *)photonURLWithSize:(CGSize)size forImageURL:(NSURL *)url
 {
-    NSString *imagePath = [NSString stringWithFormat:@"%@%@", url.host, url.path];
-    NSString *queryStr = [NSString stringWithFormat:@"resize=%.f,%.f&quality=80", size.width, size.height];
-    NSString *photonStr = [NSString stringWithFormat:@"https://i0.wp.com/%@?%@", imagePath, queryStr];
-    return [NSURL URLWithString:photonStr];
+    return [self photonURLWithSize:size forImageURL:url forceResize:YES imageQuality:DefaultPhotonImageQuality];
+}
+
++ (NSURL *)photonURLWithSize:(CGSize)size forImageURL:(NSURL *)url forceResize:(BOOL)forceResize imageQuality:(NSUInteger)quality
+{
+    // Photon will fail if the URL doesn't end in one of the accepted extensions
+    NSArray *acceptedImageTypes = @[@"gif", @"jpg", @"jpeg", @"png"];
+    if ([acceptedImageTypes indexOfObject:url.pathExtension] == NSNotFound) {
+        if (![url scheme]) {
+            return [NSURL URLWithString:[NSString stringWithFormat:@"http://%@", [url absoluteString]]];
+        }
+        return url;
+    }
+
+    NSString *urlString = [url absoluteString];
+    CGFloat scale = [[UIScreen mainScreen] scale];
+    size.width *= scale;
+    size.height *= scale;
+    quality = MIN(MAX(quality, MinPhotonImageQuality), MaxPhotonImageQuality);
+
+    // If the URL is already a Photon URL reject its photon params, and substitute our own.
+    if ([self isURLPhotonURL:url]) {
+        NSRange range = [urlString rangeOfString:@"?" options:NSBackwardsSearch];
+        if (range.location != NSNotFound) {
+            BOOL useSSL = ([urlString rangeOfString:@"ssl=1"].location != NSNotFound);
```

Comment:
Nah. This tells the Photon API that it should try requesting the specified image via https. Its not related to the scheme used to contact Photon.